lci_strategic_plugin/include/lci_strategic_plugin/lci_strategic_plugin.h

```diff
@@ -0,0 +1,621 @@
+   */
+  BoundaryDistances get_delta_x(double v0, double v1, double v_max, double v_min, double a_max, double a_min);
+  
+    /**
+   * \brief Get all possible trajectory smoothing parameters for each segments. Later this will be used to generate a single trajectory
+   * \param t current time in seconds
+   * \param v0 starting_speed
+   * \param v1 departure_speed
+   * \param v_max speed_limit
+   * \param v_min minimum speed
+   * \param a_max max_comfort_acceleration limit
+   * \param a_min max_comfort_deceleration limit
+   * \param x0 current_downtrack
+   * \param x_end traffic_light_down_track
+   * \param dx distance_remaining_to_traffic_light
+   * \param boundary_distances boundary_distances to compare against current state
+   * \return all possible trajectory smoothing parameters to later generate single trajectory
+   */
+  std::vector<TrajectoryParams> get_boundary_traj_params(double t, double v0, double v1, double v_max, double v_min, double a_max, double a_min, double x0, double x_end, double dx, BoundaryDistances boundary_distances);
+  
+  /**
+   * \brief Helper method to print TrajectoryParams
+   */
+  void print_params(TrajectoryParams params);
+
+  /**
+   * \brief Helper method to print TrajectoryParams
+   */
+  void print_boundary_distances(BoundaryDistances delta_xs);
+  
+  /**
+   * \brief Trajectory Smoothing Algorithm 8 cases.  TSMO UC2, UC3 algorithm equations
+   */
+  TrajectoryParams ts_case1(double t, double et, double v0, double v1, double v_max, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case2(double t, double et, double v0, double v1, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case3(double t, double et, double v0, double v1, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case4(double t, double et, double v0, double v1, double v_min, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case5(double t, double et, double v0, double a_max, double a_min, double x0, double x_end, double dx); //TODO Saeid: pls check dx, I included as it was missing
+  TrajectoryParams ts_case6(double t, double et, double v0, double v_min, double a_min, double x0, double x_end, double dx, double dx3, TrajectoryParams traj6);
+  TrajectoryParams ts_case7(double t, double et, double v0, double v_min, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams ts_case8(double dx, double dx5, TrajectoryParams traj8);
+
+  TrajectoryParams boundary_accel_or_decel_incomplete_upper(double t, double v0, double v1, double a_max, double a_min, double x0, double x_end, double dx);
+  TrajectoryParams boundary_accel_nocruise_notmaxspeed_decel(double t, double v0, double v1, double a_max, double a_min, double x0, double x_end, double dx); //TODO Saeid: pls check + - (which is it?) AND (a_min - a_max) will be 0 which will segfault?.
```

Comment:
Sorry for the mistake. It should be "-", not "+".

About the (a_min - a_max), it will not be zero. The assumption in all of these formulations is that a_min is negative.